x-pack/plugins/security_solution/public/common/components/header_actions/actions.tsx

```diff
@@ -234,11 +243,14 @@ const ActionsComponent: React.FC<ActionProps> = ({
     [documentBasedNotes, timelineNoteIds, securitySolutionNotesEnabled]
   );
 
-  const noteIds = useMemo(() => {
-    return securitySolutionNotesEnabled
-      ? documentBasedNotes.map((note) => note.noteId)
-      : timelineNoteIds;
-  }, [documentBasedNotes, timelineNoteIds, securitySolutionNotesEnabled]);
+  /* note ids specific to the current timeline, it is used to enable/disable pinning */
+  const noteIdsInTimeline = useMemo(() => {
```

Comment:
I think this change is a little misleading for couple of reasons:

1. These actions can run in both `timeline` and `non-timeline` environment. so `noteIdsInTimeline` will be an empty array outside timeline table ( for example, alert table ) which i think is misleading. Not your fault because there exists `timelineNoteIds` which also does the same. Although I see that `noteIdsInTimeline` is using `timelineNoteIds`, do you think there is value in combining both of them and comment the difference between them?

2. @PhilippeOberti , i think we should plan to clear/remove `eventIdToNoteIds` since that is not being used in latest when `securitySolutionNotesEnabled` is true. What do you think?

Reply:
@logeekal i think the confusion also stems from the fact that pinning and old notes are only available in timeline, but were not obvious in the actions. The complexity increases now that notes can be outside of timeline, so the "if the event has notes it must be pinned" logic no longer applies. 

I have merged the 2 variables as you suggested with added comments. I also think once the old notes system is removed, we should refactor/rewrite the pin action to have the timeline logic inside.

Reply:
Thanks @christineweng and @PhilippeOberti , I agree with both of you and leave the decision to you. Current change by @christineweng also looks pretty good to me and much more clear. Thanks 🙏 .